File: backend/internal/model/subscription.go
package model

import (
	"time"
)

type Subscription struct {
	ID             int64       `gorm:"primaryKey" json:"id"`
	TenantID       int64       `gorm:"uniqueIndex;not null" json:"tenant_id"`
	PackageID      int64       `gorm:"not null" json:"package_id"`
	PackageName    string     `gorm:"size:100" json:"package_name"`
	Status         string     `gorm:"size:20;default:'active'" json:"status"`
	StartDate      time.Time  `json:"start_date"`
	EndDate        time.Time  `json:"end_date"`
	TrialEndDate   *time.Time `json:"trial_end_date"`
	AutoRenew      int        `gorm:"default:0" json:"auto_renew"`
	MaxUsers       int        `gorm:"default:10" json:"max_users"`
	MaxStorage     int64      `gorm:"default:1073741824" json:"max_storage"`
	CurrentUsers   int        `gorm:"default:0" json:"current_users"`
	CurrentStorage int64      `gorm:"default:0" json:"current_storage"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

func (Subscription) TableName() string {
	return "subscriptions"
}

type SubscriptionInvoice struct {
	ID             int64       `gorm:"primaryKey" json:"id"`
	SubscriptionID int64       `gorm:"index;not null" json:"subscription_id"`
	TenantID       int64       `gorm:"index;not null" json:"tenant_id"`
	Amount         float64    `gorm:"type:decimal(10,2);not null" json:"amount"`
	Currency       string     `gorm:"size:10;default:'CNY'" json:"currency"`
	Status         string     `gorm:"size:20;default:'pending'" json:"status"`
	BillingPeriod  string     `gorm:"size:20" json:"billing_period"`
	PaidAt         *time.Time `json:"paid_at"`
	DueDate        time.Time  `json:"due_date"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

func (SubscriptionInvoice) TableName() string {
	return "subscription_invoices"
}

File: web/src/api/subscription.ts
import request, { type ApiResponse } from './request'

export interface Subscription {
  id: number
  tenant_id: number
  package_id: number
  package_name: string
  status: string
  start_date: string
  end_date: string
  trial_end_date: string | null
  auto_renew: number
  max_users: number
  max_storage: number
  current_users: number
  current_storage: number
  created_at: string
  updated_at: string
}

export interface SubscriptionInvoice {
  id: number
  subscription_id: number
  tenant_id: number
  amount: number
  currency: string
  status: string
  billing_period: string
  paid_at: string | null
  due_date: string
  created_at: string
  updated_at: string
}

export interface SubscriptionListResult {
  list: Subscription[]
  total: number
  page: number
  page_size: number
}

export interface InvoiceListResult {
  list: SubscriptionInvoice[]
  total: number
  page: number
  page_size: number
}

const subscriptionApi = {
  list(page = 1, pageSize = 20): Promise<ApiResponse<SubscriptionListResult>> {
    return request.get('/subscriptions', { params: { page, page_size: pageSize } })
  },

  getById(id: number): Promise<ApiResponse<Subscription>> {
    return request.get(`/subscriptions/${id}`)
  },

  create(data: Partial<Subscription>): Promise<ApiResponse<Subscription>> {
    return request.post('/subscriptions', data)
  },

  update(id: number, data: Partial<Subscription>): Promise<ApiResponse<Subscription>> {
    return request.put(`/subscriptions/${id}`, data)
  },

  delete(id: number): Promise<ApiResponse<null>> {
    return request.delete(`/subscriptions/${id}`)
  },

  cancel(id: number): Promise<ApiResponse<null>> {
    return request.post(`/subscriptions/${id}/cancel`)
  },

  renew(id: number, months: number): Promise<ApiResponse<null>> {
    return request.post(`/subscriptions/${id}/renew`, { months })
  },

  getInvoices(id: number, page = 1, pageSize = 20): Promise<ApiResponse<InvoiceListResult>> {
    return request.get(`/subscriptions/${id}/invoices`, { params: { page, page_size: pageSize } })
  }
}

export default subscriptionApi
